src/librustc_middle/ty/layout.rs

```diff
@@ -2008,7 +2012,10 @@ where
                 let tcx = cx.tcx();
                 tcx.intern_layout(Layout {
                     variants: Variants::Single { index: variant_index },
-                    fields: FieldsShape::Union(fields),
+                    fields: match NonZeroUsize::new(fields) {
+                        Some(fields) => FieldsShape::Union(fields),
+                        None => FieldsShape::Arbitrary { offsets: vec![], memory_index: vec![] },
+                        },
```

Comment:
You need to run rustfmt

Reply:
Checks passed, so this is likely a `rustfmt` bug.

We checked and it seemed like the comments just before/after `if index == variant_index &&` earlier in the larger `match` in this method, prevent `rustfmt` from reformatting the whole `match`.